genesis/recorders/recorder_manager.py

```diff
@@ -0,0 +1,182 @@
+import queue
+import threading
+from typing import TYPE_CHECKING, Callable, Type
+
+import genesis as gs
+
+if TYPE_CHECKING:
+    from .base_recorder import Recorder, RecorderOptions
+
+
+class RecorderManager:
+    """
+    Manage the creation, processing, and cleanup of all data recorders.
+
+    Parameters
+    ----------
+    step_dt: float
+        The simulation time step.
+    """
+
+    RECORDER_TYPES_MAP = {}
+
+    def __init__(self, step_dt: float):
+        self._step_dt = step_dt
+
+        self._data_funcs: list[Callable] = []
+        self._recorders: list["Recorder"] = []
+        self._data_queues: list[queue.Queue] = []
+        self._processor_threads: list[threading.Thread] = []
```

Comment:
What about adding some `Processor` class that would be associated with every single recorder?

It would store `recorder, data_queue, processor_thread, data_func`, and all the logic to for `_start_threads`, `_stop`, `_join_threads`... This would make typing much stronger if combined with `Generic`:

```python
class Processor[Generic[T]]:
    data_func: Callable[[], T]
    recorder: "Recorder"
    data_queue: queue.Queue[tuple[T, float]]
    processor_thread: threading.Thread | None
```


By the way, `Recorder` should be a "generic" class, templated by the type of data it can process.